src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -144,8 +144,10 @@ Status RedisStoreClient::AsyncGet(const std::string &table_name,
     if (!reply->IsNil()) {
       result = reply->ReadAsString();
     }
-    RAY_CHECK(!reply->IsError())
-        << "Failed to get from Redis with status: " << reply->ReadAsStatus();
```

Comment:
Let's check all callers and make sure they handle non-ok status correctly.

Reply:
Checking the caller side, it seems status propagation is properly handled.

https://github.com/ray-project/ray/blob/9a049d37a9a15fa766168c0a67feedce810d4698/src/ray/gcs/gcs_server/gcs_table_storage.cc#L39-L56

https://github.com/ray-project/ray/blob/9a049d37a9a15fa766168c0a67feedce810d4698/src/ray/gcs/gcs_server/gcs_job_manager.cc#L169-L202

I'm not worrying too much about it, since checking the function argument, especially it's `status` which indicates whether an async operation is successful is crucial.

Reply:
@jjyao I manually check all caller use cases, the only place I found problematic has updated here: https://github.com/ray-project/ray/pull/49357/commits/33f33e748fc26f0158a851cf7571a9dd7c558a39

Reply:
It seems that `AsyncGet` is also used here.

https://github.com/ray-project/ray/blob/980664e5367162bad9607ac342f5ec1e8dd9c8d9/src/ray/gcs/gcs_server/gcs_table_storage.cc#L40-L56

Reply:
Yes, I am aware of it. 

For the function you've linked, `status` is already properly handled (via propagation).
I checked the caller side as well.

Reply:
@MortalHappiness Maybe you could a look at the link I posted above.
link: https://github.com/ray-project/ray/pull/49357/files#r1893009968

Reply:
I don't think you have checked the callers. For example, here, `on_done` does not handle `status` at all.

https://github.com/ray-project/ray/blob/980664e5367162bad9607ac342f5ec1e8dd9c8d9/src/ray/gcs/gcs_server/gcs_placement_group_manager.cc#L588-L604

Reply:
https://github.com/ray-project/ray/blob/980664e5367162bad9607ac342f5ec1e8dd9c8d9/src/ray/gcs/gcs_server/gcs_placement_group_manager.cc#L715-L717

Reply:
Both fixed.